Source/Core/Core/IOS/Network/KD/NetKDRequest.cpp

```diff
@@ -217,7 +218,22 @@ NWC24::ErrorCode NetKDRequestDevice::KDDownload(const u16 entry_index,
 
   // Content metadata
   const std::string content_name = m_dl_list.GetVFFContentName(entry_index, subtask_id);
-  const std::string url = m_dl_list.GetDownloadURL(entry_index, subtask_id);
+  std::string url = m_dl_list.GetDownloadURL(entry_index, subtask_id);
+
+  // Reroute to custom server if enabled.
+  const std::vector<std::string> parts = SplitString(url, '/');
+  if (parts.size() < 3)
+  {
+    // Invalid URL
+    LogError(ErrorType::KD_Download, NWC24::WC24_ERR_SERVER);
+    return NWC24::WC24_ERR_SERVER;
+  }
+
+  if (std::optional<std::string> patch =
+          WC24PatchEngine::GetNetworkPatch(parts[2], WC24PatchEngine::IsKD{true}))
+  {
+    url = ReplaceAll(url, parts[2], patch.value());
```

Comment:
This seems too broad, this would also replace instances outside the domain part.